network/network_test.go

```diff
@@ -282,6 +282,11 @@ func newFullyConnectedTestNetwork(t *testing.T, handlers []router.InboundHandler
 		if i != 0 {
 			config := configs[0]
 			net.ManuallyTrack(config.MyNodeID, config.MyIPPort.Get())
+			// Wait until the node is connected to the first node.
+			// This forces nodes to connect to each other in a deterministic order.
+			require.Eventually(func() bool {
+				return len(net.PeerInfo([]ids.NodeID{config.MyNodeID})) > 0
+			}, time.Minute, time.Millisecond)
```

Comment:
nit: On the success case, it would typically take 10-50ms before this statement completes.
On the fail case, it would take at least one minute. Would you consider dial the fail case threshold down to 10 seconds ?